pkg/ibm/resources_container.go

```diff
@@ -157,6 +158,34 @@ func (resources *ResourcesContainer) collectTags() error {
 	return nil
 }
 
+func (resources *ResourcesContainer) verifyResourceGroupID(apiKey string) error {
+	rm, err := resourcemanagerv2.NewResourceManagerV2(&resourcemanagerv2.ResourceManagerV2Options{
+		Authenticator: &core.IamAuthenticator{
+			ApiKey: apiKey,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("error creating resource manager service: %w", err)
+	}
+
+	resourceGroup, _, err := rm.GetResourceGroup(rm.NewGetResourceGroupOptions(
+		resources.resourceGroupID,
+	))
+	if err == nil && resourceGroup != nil {
+		return nil
+	}
+
+	listResourceGroupsOptions := rm.NewListResourceGroupsOptions()
+	listResourceGroupsOptions.SetName(resources.resourceGroupID)
+
+	resourceGroupList, _, err := rm.ListResourceGroups(&resourcemanagerv2.ListResourceGroupsOptions{Name: &resources.resourceGroupID})
+	if err == nil && len(resourceGroupList.Resources) == 1 {
+		resources.resourceGroupID = *resourceGroupList.Resources[0].ID
+		return nil
+	}
```

Comment:
```suggestion
	if err == nil && resourceGroup != nil {
		return nil // user provided us with a valid resource group ID
	}

    // check if the user provided us with the resource group name rather than its id
	listResourceGroupsOptions := rm.NewListResourceGroupsOptions()
	listResourceGroupsOptions.SetName(resources.resourceGroupID)

	resourceGroupList, _, err := rm.ListResourceGroups(&resourcemanagerv2.ListResourceGroupsOptions{Name: &resources.resourceGroupID})
	if err == nil && len(resourceGroupList.Resources) == 1 {
		resources.resourceGroupID = *resourceGroupList.Resources[0].ID
		return nil // user provided us with a valid resource group name
	}
```